### ISSUE TITLE
Integration fails to load on HA 2024.6.0b6 (beta)

### ISSUE THREAD

**Issue by sparkydave1981** [bkbilly/tpms_ble#4]
I updated my HA to the 2024.6 beta and this integration now fails to load, giving the below error in the log.

I'm on HAOS 12.3 and Supervisor 2024.06.0

```
Logger: homeassistant.config_entries
Source: config_entries.py:594
First occurred: 09:34:12 (2 occurrences)
Last logged: 16:32:20

Error setting up entry TPMS C7C3 for tpms_ble
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1249, in _load_platform
    cache[full_name] = self._import_platform(platform_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1281, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 131, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/tpms_ble/sensor.py", line 117, in <module>
    PassiveBluetoothDataProcessor[Optional[Union[str, int]]]
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/typing.py", line 384, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/typing.py", line 1066, in _generic_class_getitem
    _check_generic(cls, params, len(cls.__parameters__))
  File "/usr/local/lib/python3.12/site-packages/typing_extensions.py", line 2947, in _check_generic
    raise TypeError(f"Too {'many' if alen > elen else 'few'} arguments"
TypeError: Too few arguments for <class 'homeassistant.components.bluetooth.passive_update_processor.PassiveBluetoothDataProcessor'>; actual 1, expected at least 2

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 594, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/tpms_ble/__init__.py", line 37, in async_setup_entry
    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2031, in async_forward_entry_setups
    await integration.async_get_platforms(platforms)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1162, in async_get_platforms
    import_future.result()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1150, in async_get_platforms
    platforms.update(self._load_platforms(platform_names))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1075, in _load_platforms
    platform_name: self._load_platform(platform_name)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1269, in _load_platform
    raise ImportError(
ImportError: Exception importing custom_components.tpms_ble.sensor
```

I'm not sure how much help that is...

**Comment by bkbilly** [comment]
It seems that the data in the PassiveBluetoothProcessorEntity is not required, so you could remove it entirely.
The new block should look like this:
```python
class TPMSBluetoothSensorEntity(
    PassiveBluetoothProcessorEntity,
    SensorEntity,
):

```

Please let me know if it works on your version of Home Assistant.

**Comment by sparkydave1981** [comment]
This worked perfectly, thank you!  I didn't try your first suggestion, only the second one (I was sleeping).  I've since updated from the 2024.6 beta to 2024.6.0 stable with this code change working so I'd say you should be fine to include it in an update to the integration. 

Thanks again for your great work mate.

Cheers,

Dave

**Comment by bkbilly** [comment]
Thank you for trying it out and letting me know about this.
I've released a new version with the fix to this issue.

**Comment by jdeath** [comment]
I had the same issue with one of my integrations. I created an issue as I thought it was an undocumented breaking change, but they told me I was wrong to use the `PassiveBluetoothDataProcessor[Optional[Union[str, int]]]` syntax in the first place (though I copied from a core integration originally!). I figured out the first work around by seeing what changes they made in core components. I'll try your workaround next.
